### PR TITLE
Correct diff map

### LINKED ISSUE #1290: Glitches in diff_map ...
Apparently the diff_map application does not accept the json configuration file.
Moreover it is not possible to re-load such files neither

### PR DESCRIPTION
**Pull request by kif**
Correct many glitches in the diff_map application
close #1290

### CODE REVIEW

File: pyFAI/diffmap.py [hunk 1]
```diff
@@ -371,11 +374,11 @@ def setup_ai(self):
                           " no poni file provided"))
             raise RuntimeError("You must provide poni a file")
         if self.dark:
-            self.ai.set_darkfiles(self.dark)
+            self.ai.detector.set_darkcurrent(_reduce_images(self.dark))
         if self.flat:
-            self.ai.set_flatfiles(self.flat)
+            self.ai.detector.set_darkcurrent(_reduce_images(self.flat))
```
Comment (vallsv) [inline pyFAI/diffmap.py]: Is that a typo? dark/flat

File: pyFAI/gui/diffmap_widget.py [hunk 2]
```diff
@@ -440,19 +447,20 @@ def process(self, config=None):
         t0 = time.time()
         with self.processing_sem:
             config = self.dump()
+            print(config)
```
Comment (vallsv) [inline pyFAI/gui/diffmap_widget.py]: In case, could be removed?

File: pyFAI/gui/diffmap_widget.py [hunk 3]
```diff
@@ -440,19 +447,20 @@ def process(self, config=None):
...
                               npt_azim=config_ai.get("nbpt_azim", 1) if config_ai.get("do_2D") else None)
             diffmap.inputfiles = [i.path for i in self.list_dataset]  # in case generic detector without shape
-            diffmap.ai = worker.make_ai(config_ai)
-            # TODO: This diffmap configuration file should be cleaned up
-            reader = ConfigurationReader(config_ai)
-            diffmap.method = reader.pop_method("csr")
-            diffmap.unit = to_unit(config_ai.get("unit", "2th_deg"))
+            diffmap.worker = worker.Worker()
+            diffmap.worker.set_config(config_ai, consume_keys=False)
+            print(diffmap.worker.get_config())
```
Comment (vallsv) [inline pyFAI/gui/diffmap_widget.py]: In case, could be removed?

File: pyFAI/worker.py [hunk 4]
```diff
@@ -538,6 +507,7 @@ def set_config(self, config, consume_keys=False):
         apply_values = config.pop("do_radial_range", True)
         if apply_values and value1 is not None and value2 is not None:
             self.radial_range = float(value1), float(value2)
+            print("radial range", self.radial_range)
```
Comment (vallsv) [inline pyFAI/worker.py]: In case, could be removed?

File: pyFAI/worker.py [hunk 5]
```diff
@@ -459,6 +427,7 @@ def set_config(self, config, consume_keys=False):
         :param bool consume_keys: If true the keys from the dictionary will be
             consumed when used.
         """
+        print("set_config", config)
```
Comment (vallsv) [inline pyFAI/worker.py]: In case, could be removed?

**Comment by vallsv** [comment]
I did some comments, only the first looks important. Except that, it looks fine.

File: pyFAI/diffmap.py (hunk 1 above)
Comment (kif) [reply pyFAI/diffmap.py]: thanks for spotting this one !

**Pull request merged**